### PR TITLE
Added support for the Python 3.5 matrix multiplication operator.

### PR DESCRIPTION
**Pull request by moorepants**
Python 3.5 introduced the @ operator for matrix multiplication. This adds
support for the operator to our matrix types. It mirrors the multiplication
operator *.

### CODE REVIEW

File: sympy/matrices/tests/test_matrices.py [hunk 1]
```diff
@@ -128,6 +128,15 @@ def test_multiplication():
     assert c[2, 0] == 18
     assert c[2, 1] == 0
 
+    if PY3:
```
Comment (moorepants) [inline sympy/matrices/tests/test_matrices.py]: This should check for > 3.5.

Comment (asmeurer) [inline sympy/matrices/tests/test_matrices.py]: This is invalid syntax in < 3.5. You will have to test the operator itself using `eval`.

**Comment by asmeurer** [comment]
The 3.5-only testing needs to be fixed, but otherwise this looks good.

File: sympy/matrices/sparse.py [hunk 2]
```diff
@@ -395,6 +395,9 @@ def __mul__(self, other):
             return other._new(self*self._new(other))
         return self.scalar_multiply(other)
 
+    def __matmul__(self, other):
```
Comment (asmeurer) [inline sympy/matrices/sparse.py]: You should be able to do just `__matmul__ = __mul__`.

Comment (moorepants) [inline sympy/matrices/sparse.py]: Thanks. I'll fix.

**Comment by moorepants** [comment]
How should I avoid the syntax errors when importing the test files in Python < 3.5?

File: sympy/matrices/tests/test_matrices.py (hunk 1 above)
Comment (moorepants) [inline sympy/matrices/tests/test_matrices.py]: Ah, yes, that will work.

**Comment by moorepants** [comment]
Ok, I think this works now.

**Comment by asmeurer** [comment]
3.5 tests aren't actually run in Travis until https://github.com/sympy/sympy/pull/10084 is merged.

**Comment by moorepants** [comment]
It runs locally for me on 3.5. Test pass. We can wait till #10084 is merged if you want.

**Comment by asmeurer** [comment]
That PR is ready to go (the final Travis run still needs to finish). I might wait a bit to see if anyone wants to review it, and if not, I will just merge it.

**Comment by moorepants** [comment]
Shall we merge this now?

**Comment by asmeurer** [comment]
You might want to merge in master so that Travis runs the tests on 3.5.

**Comment by moorepants** [comment]
Done.

**Comment by asmeurer** [comment]
+1 to merge when the tests finish

**Pull request merged**